src/editor/Core/Reducer.re

```diff
@@ -73,10 +73,13 @@ let reduce: (State.t, Actions.t) => State.t =
         buffers: BufferMap.updateMetadata(s.buffers, bs.buffers),
         tabs: showTablineBuffers(s, bs.buffers),
       }
-    | BufferUpdate(bu) => {
+    | BufferUpdate(bu) =>
+      let r = {
         ...s,
         buffers: BufferMap.update(bu.id, applyBufferUpdate(bu), s.buffers),
```

Comment:
Hmm, ya, it looks like when we open a new file it comes in with a buffer update at `start: 0` and `end: -1`. Maybe we aren't handling this case properly in `Buffer.applyUpdate`, if there is existing content in the buffer - might be worth adding a test?

Reply:
Yep seems we were trying to update the buffer in this case, so now we check for that case and if it occurs we set the update lines (aka the buffer content) also added a test case